pkg/odo/util/validation/validation.go

```diff
@@ -0,0 +1,22 @@
+package validation
+
+import (
+	"fmt"
+	"strings"
+
+	"k8s.io/apimachinery/pkg/util/validation"
+)
+
+// ValidateName will do validation of application & component names according to DNS (RFC 1123) rules
+// Criteria for valid name in kubernetes: https://github.com/kubernetes/community/blob/master/contributors/design-proposals/architecture/identifiers.md
+func ValidateName(name string) error {
```

Comment:
was thinking if this can be one of the validators
but I somehow have strong dislike(may be bcoz I have always got myself strangled in type assertions and conversions) for golang interface{} but where necessary they have to be used :)

Reply:
with go, I have always more enjoyed things being more typed than generic may be bcoz I felt that was the go principles but I might be wrong :)

Reply:
> was thinking if this can be one of the validators

I see its done below